RUFAS/routines/animal/life_cycle/animal_population.py

```diff
@@ -0,0 +1,207 @@
+from dataclasses import dataclass
+from random import shuffle
+from typing import List, Dict
+
+from RUFAS.routines.animal.life_cycle.calf import Calf
+from RUFAS.routines.animal.life_cycle.cow import Cow
+from RUFAS.routines.animal.life_cycle.heiferI import HeiferI
+from RUFAS.routines.animal.life_cycle.heiferII import HeiferII
+from RUFAS.routines.animal.life_cycle.heiferIII import HeiferIII
+
+
+@dataclass(kw_only=True)
+class AnimalPopulation:
+    """
+    A data class representing the population of animals in a herd.
+
+    Attributes
+    ----------
+    calves : List[Calf]
+        A list of Calf instances in the herd.
+    heiferIs : List[HeiferI]
+        A list of HeiferI (stage I heifers) instances in the herd.
+    heiferIIs : List[HeiferII]
+        A list of HeiferII (stage II heifers) instances in the herd.
+    heiferIIIs : List[HeiferIII]
+        A list of HeiferIII (stage III heifers) instances in the herd.
+    cows : List[Cow]
+        A list of Cow instances in the herd.
+    replacement : List[Cow]
+        A list of replacement Cow instances in the herd.
+    current_animal_id : int, optional
+        The highest ID number among all animals in the herd. Default is 0.
+    order_by_random : bool, optional
+        A flag to indicate whether the animals should be ordered randomly.
+        Default is True.
+    """
+    calves: List[Calf]
+    heiferIs: List[HeiferI]
+    heiferIIs: List[HeiferII]
+    heiferIIIs: List[HeiferIII]
+    cows: List[Cow]
+    replacement: List[Cow]
+
+    current_animal_id: int = 0
+    order_by_random: bool = True
+
+    def __post_init__(self):
+        """Post init function to find the max id of all animals, and set the current_animal_id"""
+        ids = [i.id for i in self.calves] + [i.id for i in self.heiferIs] + [i.id for i in self.heiferIIs] + \
+              [i.id for i in self.heiferIIIs] + [i.id for i in self.cows] + [i.id for i in self.replacement]
```

Comment:
```suggestion
        ids = [i.id for i in self.calves + self.heiferIs + self.heiferIIs + self.heiferIIIs + self.cows + 
               self.replacement]
```
Slightly more pythonic IMO, but doesn't actually reduce the number of lines.